MapboxNavigation/VisualInstructionComponent.swift

```diff
@@ -5,16 +5,20 @@ extension VisualInstructionComponent {
     
     static let scale = UIScreen.main.scale
     
-    func cacheKey() -> String? {
+    var cacheKey: String? {
         switch type {
         case .exit, .exitCode:
             guard let exitCode = self.text else { return nil}
             return "exit-" + exitCode + "-\(VisualInstructionComponent.scale)-\(hashValue)"
-        case .image, .text:
-            guard let imageURL = imageURL else { return nil }
+        case .image:
+            guard let imageURL = imageURL else { return genericCacheKey }
             return "\(imageURL.absoluteString)-\(VisualInstructionComponent.scale)"
-        case .delimiter:
+        default:
             return nil
         }
     }
+    
+    var genericCacheKey: String {
+        return "generic-" + (text ?? "nil")
```

Comment:
Are there cases we could end up with different shields that have the same cache key, ie `generic-` because text is nil?

Reply:
A good test case might be, two different steps where the number is given but no text associated with it? Maybe it can't happen but worth checking out.

Reply:
@bsudekum The `InstructionPresenter` doesn't try to render Generic Shields with no text, as they would be an empty box.